### PR TITLE
Development

### PR DESCRIPTION
**Pull request by igasm**
- changes in POMs
- changes in test for server, added tests
- javaDoc for common module

### CODE REVIEW

File: common/src/main/java/battleships/communication/ClientHandlerBuilder.java [hunk 1]
```diff
@@ -2,26 +2,47 @@
...
+/**
+ * This is a builder for ClientHandler class.
+ */
 public class ClientHandlerBuilder {
   private MessageReceiver messageReceiver;
   private MessageSender messageSender;
   private Socket socket;
 
+  /**
+   * This method should be used as first in ClientHandlerBuilder chain.
```
Comment (EmiliaCiastek) [inline common/src/main/java/battleships/communication/ClientHandlerBuilder.java]: if socket is mandatory constructor of ClientHandlerBuilder should accept it as a parameter

File: common/src/main/java/battleships/communication/ClientHandlerBuilder.java [hunk 2]
```diff
@@ -2,26 +2,47 @@
...
+   */
   public ClientHandlerBuilder addMessageSender() {
     messageSender = new MessageSenderBuilder().addSocket(socket).build();
     return this;
   }
 
+  /**
+   * It returns new instance of ClientHandler class.
+   */
   public ClientHandler build() {
```
Comment (EmiliaCiastek) [inline common/src/main/java/battleships/communication/ClientHandlerBuilder.java]: maybe use something like in FluentAPI?

File: common/src/main/java/battleships/communication/jsonhandlers/JsonMarshaller.java [hunk 3]
```diff
@@ -5,14 +5,23 @@
 import battleships.utils.BattleshipUtils;
 import com.fasterxml.jackson.core.JsonProcessingException;
 
+/**
+ * This class is a Json implementation of Marshaller. Its main responsibility is to turn a Messageable
+ * object into its Json String representation using MessageableMapper wrapper class.
```
Comment (EmiliaCiastek) [inline common/src/main/java/battleships/communication/jsonhandlers/JsonMarshaller.java]: wrapper?

File: common/src/test/java/battleships/communication/ClientHandlerTest.java [hunk 4]
```diff
@@ -24,7 +24,7 @@ public void setUp() throws Exception {
   @Test
   public void whenClientHandlerSendMessage_expectMessageSenderInvokeMethodOnce() throws Exception {
     //when
-    clientHandler.sendMessage(new GoodByeMessage("good bye!"));
+    clientHandler.sendMessage(new WelcomeMessage("Welcome!"));
     //then
     assertThat(mockingDetails(messageSenderMock).getInvocations().size()).isEqualTo(1);
```
Comment (EmiliaCiastek) [inline common/src/test/java/battleships/communication/ClientHandlerTest.java]: use verify (Mockito)

File: server/src/main/java/battleships/gameplay/CalculatingGameResult.java [hunk 5]
```diff
@@ -39,25 +40,32 @@ private boolean calculateGameResult(List<Fleet> playersFleets, List<SalvoResult>
       results.forEach(p -> p.setGameResult(GameResult.DRAW));
       log.info("We have a draw");
       return true;
+
     } else if (sunkenFleets.size() == 1) {
```
Comment (EmiliaCiastek) [inline server/src/main/java/battleships/gameplay/CalculatingGameResult.java]: sunken???

File: server/src/main/java/battleships/gameplay/SendingSalvoResults.java [hunk 6]
```diff
@@ -22,7 +22,9 @@
 
   @Override
   public GameState process() {
-    IntStream.range(0, observers.size()).forEach(p -> observers.get(p).sendMessage(results.get(p)));
+    IntStream
+        .range(0, observers.size())
+        .forEach(p -> observers.get(p).sendMessage(results.get(p)));
```
Comment (EmiliaCiastek) [inline server/src/main/java/battleships/gameplay/SendingSalvoResults.java]: hmmm....

File: server/src/main/java/battleships/gameplay/SinkingShips.java [hunk 7]
```diff
@@ -19,15 +21,24 @@
...
+   * @return new CalculatingGameResult state.
+   */
   @Override
   public GameState process() {
-    killShips(playersFleets.get(0).getShips(), results.get(1));
-    killShips(playersFleets.get(1).getShips(), results.get(0));
+    AtomicInteger resultIndex = new AtomicInteger(RESULT_MAXIMUM_INDEX);
+
+    playersFleets
+        .forEach(p -> killShips(p.getShips(), results.get(resultIndex.getAndDecrement())));
```
Comment (EmiliaCiastek) [inline server/src/main/java/battleships/gameplay/SinkingShips.java]: "Delete this", Krzysztof said

File: server/src/test/java/battleships/gameplay/EndingGameTest.java [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+  private EndingGame endingGame;
+
+  @BeforeMethod
+  public void setUp() {
+    endingGame = new EndingGame();
+  }
+
+  @Test
+  public void whenInEndOfGameState_expectThatOnlyEndOfGameStateWillBeReturned() {
+    assertThat(endingGame.process()).isExactlyInstanceOf(EndingGame.class);
```
Comment (EmiliaCiastek) [inline server/src/test/java/battleships/gameplay/EndingGameTest.java]: weird

File: server/src/test/java/battleships/gameplay/Triplet.java [hunk 9]
```diff
@@ -0,0 +1,144 @@
...
+  List<SalvoResult> salvoResults;
+
+  private Triplet(List<Observers> observers, List<Fleet> fleets, List<SalvoResult> salvoResults) {
+    this.observers = observers;
+    this.fleets = fleets;
+    this.salvoResults = salvoResults;
+  }
+
+  static Object[][] provideTestDataForSinkingShips() {
+    return new Object[][] {
```
Comment (EmiliaCiastek) [inline server/src/test/java/battleships/gameplay/Triplet.java]: magic numbers

File: common/src/main/java/battleships/communication/ClientHandlerBuilder.java (hunk 1 above)
Comment (igasm) [reply common/src/main/java/battleships/communication/ClientHandlerBuilder.java]: Robert is working on that

File: common/src/main/java/battleships/communication/ClientHandlerBuilder.java (hunk 2 above)
Comment (igasm) [reply common/src/main/java/battleships/communication/ClientHandlerBuilder.java]: i created issue for that: 
ClientHandlerBuilder - fluentAPI #95

File: common/src/main/java/battleships/communication/jsonhandlers/JsonMarshaller.java (hunk 3 above)
Comment (igasm) [reply common/src/main/java/battleships/communication/jsonhandlers/JsonMarshaller.java]: changed

File: common/src/test/java/battleships/communication/ClientHandlerTest.java (hunk 4 above)
Comment (igasm) [reply common/src/test/java/battleships/communication/ClientHandlerTest.java]: Robert is working on that

File: server/src/main/java/battleships/gameplay/CalculatingGameResult.java (hunk 5 above)
Comment (igasm) [reply server/src/main/java/battleships/gameplay/CalculatingGameResult.java]: changed

File: server/src/main/java/battleships/gameplay/SinkingShips.java (hunk 7 above)
Comment (igasm) [reply server/src/main/java/battleships/gameplay/SinkingShips.java]: he changed his mind :)

File: server/src/test/java/battleships/gameplay/EndingGameTest.java (hunk 8 above)
Comment (igasm) [reply server/src/test/java/battleships/gameplay/EndingGameTest.java]: no idea, how to make it better

**Pull request merged**